azurerm/internal/services/network/firewall_network_rule_collection_resource.go

```diff
@@ -473,6 +485,9 @@ func flattenFirewallNetworkRuleCollectionRules(rules *[]network.AzureFirewallNet
 		if rule.DestinationPorts != nil {
 			output["destination_ports"] = set.FromStringSlice(*rule.DestinationPorts)
 		}
+		if rule.DestinationFqdns != nil {
+			output["destination_fqdns"] = set.FromStringSlice(*rule.DestinationFqdns)
+		}
```

Comment:
this (and the other blocks) need to be output as an empty list in the case they're not returned from the API to be able to correctly map the diff, so could we make this:

```suggestion
		destinationFQDNs := make([]string, 0)
		if rule.DestinationFqdns != nil {
			destinationFQDNs = set.FromStringSlice(*rule.DestinationFqdns)
		}
		output["destination_fqdns"] = destinationFQDNs
```